src/nix/develop.cc

```diff
@@ -255,6 +255,11 @@ struct Common : InstallableCommand, MixProfile
         // FIXME: properly unquote 'outputs'.
         StringMap rewrites;
         for (auto & outputName : tokenizeString<std::vector<std::string>>(replaceStrings(outputs->second.quoted, "'", ""))) {
+            std::regex ptrn(R"re(\[([A-z0-9]+)\]=.*)re");
+            std::smatch match;
+            if (std::regex_match(outputName, match, ptrn)) {
+                outputName = match[1];
+            }
```

Comment:
That’s probably worth a comment, because right out of the blue this doesn’t make any sense to me 🤔